### PR TITLE
Modernization/updating c++ pass 1

### PR DESCRIPTION
**Pull request by SeanAlling-DojoFive**
First pass addressing Issue #41. 

Pushing branch to to get exposure on changes.

### CODE REVIEW

**Comment by sierrafoxtrot** [comment]
Thanks @SeanAlling-DojoFive, this is looking great. I'm hoping to get through the review today. 

If there are no questions/issues, are you happy if I squash-merge? I'm still getting the lay of the land on what is considered good form on GH for merging contributions and to me this one looks like a hefty but single update if that makes sense.

**Comment by SeanAlling-DojoFive** [comment]
That is fine, I am still looking at some cleanup and verifying this are all correct.

Should have my changes done this weekend and then I’ll do a squash merge.

**Comment by sierrafoxtrot** [comment]
No worries, take your time. I'll kick off when you think it's ready.

**Comment by SeanAlling-DojoFive** [comment]
Okay, I have gone through every file and attempted to move all default member functions to be in class. @sierrafoxtrot I think now is a good time to start taking a look. 

My ultimate goal of this first PR is to address update all constructors. Other PR's will focus on other C++11 modernization's.

**Comment by sierrafoxtrot** [comment]
@SeanAlling-DojoFive , it's might be worth noting that I need to manually approve the running of the checks which is no doubt a pain (I can relax this setting but not for recently created GH accounts :-( ). We likely have a timezone difference which is slowing things down for you. If you submit a separate smaller commit, GH will no longer consider you a first-time-committer and the checks should trigger automatically speeding up the cycle.

**Comment by sierrafoxtrot** [comment]
Hey @SeanAlling-DojoFive , I'm getting through this a bit each day. Confident that I'll have it wrapped on the weekend.

File: srecord/memory/walker/alignment.h [hunk 1]
```diff
@@ -76,25 +76,27 @@ class memory_walker_alignment:
     memory_walker_alignment(unsigned multiple);
 
     unsigned multiple;
```
Comment (sierrafoxtrot) [inline srecord/memory/walker/alignment.h:78]: Suggest adding a default member init value for completeness

**Review by sierrafoxtrot — CHANGES_REQUESTED** [review]
Thank you Sean for this extensive contribution. My apologies for the time taken to respond but I wanted to do it justice. A couple of minor comments.

**Review by sierrafoxtrot — APPROVED** [review]

**Pull request merged**